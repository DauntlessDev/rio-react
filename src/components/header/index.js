import React, { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
    Background, Container, Feature, Text, Link as TextLink, ButtonLink, Logo, Darken, Inner, Group, FeatureCallOut, FeatureCallOutBottom, Dropdown,
    SearchIcon,
    SearchInput, Search, SubText,
    Picture, Profile, PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
    return bg ? (
        <Background {...restProps} data-testid="header-bg">
            <Darken>
                {children}
            </Darken>
        </Background>
    ) : <>{children}</>
}


Header.Profile = function HeaderProfile({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>;
};


Header.Feature = function HeaderFeature({ children, ...restProps }) {
    return <Feature {...restProps}>{children}</Feature>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}
Header.FeatureCallOutBottom = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOutBottom {...restProps}>{children}</FeatureCallOutBottom>
}

Header.Text = function HeaderText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}

Header.SubText = function HeaderText({ children, ...restProps }) {
    return <SubText {...restProps}>{children}</SubText>
}


Header.Frame = function HeaderFrame({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}


Header.Inner = function HeaderInner({ children, ...restProps }) {
    return <Inner {...restProps}>{children}</Inner>
}



Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>
}



Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
    return <TextLink {...restProps}>{children}</TextLink>
}



Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReactRouterLink to={to}><Logo {...restProps}></Logo></ReactRouterLink>
    )
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
    const [searchActive, setSearchActive] = useState(false);

    return (
        <Search {...restProps}>
            <SearchIcon onClick={() => setSearchActive((searchActive) => !searchActive)} data-testid="search-click">
                <img src="/images/icons/search.png" alt="Search" />
            </SearchIcon>
            <SearchInput
                value={searchTerm}
                onChange={({ target }) => setSearchTerm(target.value)}
                placeholder="Search films and series"
                active={searchActive}
                data-testid="search-input"
            />
        </Search>
    );
};


Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
    return <PlayButton {...restProps}>{children}</PlayButton>;
};

