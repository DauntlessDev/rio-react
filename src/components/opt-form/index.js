import React from 'react'
import { Container, Input, Button, Text } from './styles/opt-form'

export default function OptForm({ children, ...restProps }) {
    return (
        <Container{...restProps}>
            {children}
        </Container>
    )
}

OptForm.Input = function OutFormInput({ ...restProps }) {
    return <Input {...restProps} />
}

OptForm.Button = function OutFormButton({ children, ...restProps }) {
    return (
        <Button {...restProps}>
            {children}
            <img src="/images/icons/chevron-right.png" alt="Try Now" />
        </Button>
    )
}

OptForm.Text = function OutFormText({ children, ...restProps }) {
    return (
        <Text {...restProps}>
            {children}
        </Text>
    )
}



OptForm.Break = function OutFormBreak({ ...restProps }) {
    return (
        <Text {...restProps} />
    )
}

