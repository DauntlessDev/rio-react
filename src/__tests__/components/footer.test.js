import React from 'react'
import { render } from '@testing-library/react'
import { Footer } from '../../components'


describe('<Footer/>', () => {
    it('renders the <Footer /> with a popuolated data', () => {
        const { container, getByText } = render(
            <Footer>
                <Footer.Title>Questions? Contact Us.</Footer.Title>
                <Footer.Break />
                <Footer.Row>
                    <Footer.Column>
                        <Footer.Link href="#">FAQ</Footer.Link>
                        <Footer.Link href="#">Investor Relations</Footer.Link>
                        <Footer.Link href="#">Ways to Watch</Footer.Link>
                        <Footer.Link href="#">Corporate Information</Footer.Link>
                        <Footer.Link href="#">Netflix Originals</Footer.Link>
                    </Footer.Column>


                    <Footer.Column>
                        <Footer.Link href="#">Help Centre</Footer.Link>
                        <Footer.Link href="#">Jobs</Footer.Link>
                        <Footer.Link href="#">Terms of Use</Footer.Link>
                        <Footer.Link href="#">Contact Us</Footer.Link>
                    </Footer.Column>


                    <Footer.Column>
                        <Footer.Link href="#">Account</Footer.Link>
                        <Footer.Link href="#">Redeem Gift Cards</Footer.Link>
                        <Footer.Link href="#">Privacy</Footer.Link>
                        <Footer.Link href="#">Speed Test</Footer.Link>
                    </Footer.Column>

                    <Footer.Column>
                        <Footer.Link href="#">Media Centre</Footer.Link>
                        <Footer.Link href="#">Buy Gift Cards</Footer.Link>
                        <Footer.Link href="#">Cookie Preferences</Footer.Link>
                        <Footer.Link href="#">Lagal Notice</Footer.Link>
                    </Footer.Column>
                </Footer.Row>
                <Footer.Break />
                <Footer.Text>Rio Philippines</Footer.Text>
            </Footer>
        )

        expect(getByText('Questions? Contact Us.')).toBeTruthy();
        expect(getByText('FAQ')).toBeTruthy();
        expect(getByText('Investor Relations')).toBeTruthy();
        expect(getByText('Ways to Watch')).toBeTruthy();
        expect(getByText('Corporate Information')).toBeTruthy();
        expect(getByText('Netflix Originals')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    })
})