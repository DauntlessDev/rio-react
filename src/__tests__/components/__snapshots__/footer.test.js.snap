// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders the <Footer /> with a popuolated data 1`] = `
<div
  class="footer__Container-sc-1qr3430-0 ckFPBR"
>
  <p
    class="footer__Title-sc-1qr3430-4 hWytSb"
  >
    Questions? Contact Us.
     
  </p>
  <p
    class="footer__Break-sc-1qr3430-6 lnQrXy"
  >
     
  </p>
  <div
    class="footer__Row-sc-1qr3430-2 gtGiYx"
  >
    <div
      class="footer__Column-sc-1qr3430-1 cjxaMq"
    >
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        FAQ
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Investor Relations
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Ways to Watch
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Corporate Information
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Netflix Originals
         
      </a>
       
    </div>
    <div
      class="footer__Column-sc-1qr3430-1 cjxaMq"
    >
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Help Centre
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Jobs
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Terms of Use
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Contact Us
         
      </a>
       
    </div>
    <div
      class="footer__Column-sc-1qr3430-1 cjxaMq"
    >
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Account
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Redeem Gift Cards
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Privacy
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Speed Test
         
      </a>
       
    </div>
    <div
      class="footer__Column-sc-1qr3430-1 cjxaMq"
    >
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Media Centre
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Buy Gift Cards
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Cookie Preferences
         
      </a>
      <a
        class="footer__Link-sc-1qr3430-3 hieEbN"
        href="#"
      >
        Lagal Notice
         
      </a>
       
    </div>
     
  </div>
  <p
    class="footer__Break-sc-1qr3430-6 lnQrXy"
  >
     
  </p>
  <p
    class="footer__Text-sc-1qr3430-5 cbOTXu"
  >
    Rio Philippines
     
  </p>
</div>
`;
